### PR TITLE
Added the running rosa spots ability

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Span the Rosa worker nodes with the help of machinepools by creating the default cluster with 3(M+W+I). 

## Related Tickets & Documents
Reference document: [Link](https://docs.openshift.com/rosa/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.html#creating_machine_pools_cli_rosa-managing-worker-nodes)

- Related Issue #
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change. 
  - Add the variable enable_spot_workers=true  in the install config file of rosa.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.
   - Ran the airflow cluster by creating a [playground](https://athiruma-playground-thiru-airflow.apps.sailplane.perf.lab.eng.rdu2.redhat.com/dags/4.13-rosa-sts-ovn-small-cp/grid?tab=logs&dag_run_id=manual__2023-08-01T03%3A21%3A15.546229%2B00%3A00&task_id=node-density) on the sailplane cluster.

### CODE REVIEW

**Comment by krishvoor** [comment]
@athiruma Can you share the Airflow DAG for review?

**Comment by athiruma** [comment]
> @athiruma Can you share the Airflow DAG for review?

yes, it is attached to the word playground in the above description.

File: dags/openshift_nightlies/config/install/rosa/ovn-small.json [hunk 1]
```diff
@@ -12,5 +12,6 @@
     "openshift_network_type": "OVNKubernetes",
     "openshift_worker_instance_type": "m5.2xlarge",
     "machineset_metadata_label_prefix": "machine.openshift.io",
-    "openshift_workload_node_instance_type": null
+    "openshift_workload_node_instance_type": null,
+    "enable_spot_workers": false
```
[the remark this answers is not in the document]
Comment (krishvoor) [reply dags/openshift_nightlies/config/install/rosa/ovn-small.json:16]: Add the `openshift_workload_node_instance_type`

File: dags/openshift_nightlies/scripts/install/rosa.sh [hunk 2]
```diff
@@ -511,6 +521,7 @@ setup(){
     fi
 }
 
+
```
Comment (krishvoor) [inline dags/openshift_nightlies/scripts/install/rosa.sh:524]: Minor nit, extra space

**Review by krishvoor — COMMENTED** [review]
Minor suggestions, overall lgtm

**Comment by krishvoor** [comment]
> > @athiruma Can you share the Airflow DAG for review?
> 
> yes, it is attached to the word playground in the above description.

Ahh, I see it now thank you!

File: dags/openshift_nightlies/tasks/install/rosa/defaults.json [hunk 3]
```diff
@@ -41,5 +41,6 @@
     "staging_mgmt_provisioner_shards": "",
     "aws_region": "us-west-2",
     "oidc_config": "",
-    "extra_machinepool": []
+    "extra_machinepool": [],
+    "enable_spot_workers": false
```
[the remark this answers is not in the document]
Comment (athiruma) [reply dags/openshift_nightlies/tasks/install/rosa/defaults.json:45]: Default , there is no variable with that name

[the remark this answers is not in the document]
Comment (krishvoor) [reply dags/openshift_nightlies/tasks/install/rosa/defaults.json:45]: You can declare new variable with empty/null values in `defaults.json` file

[the remark this answers is not in the document]
Comment (athiruma) [reply dags/openshift_nightlies/tasks/install/rosa/defaults.json:45]: ok

File: dags/openshift_nightlies/config/install/rosa/ovn-small.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (athiruma) [reply dags/openshift_nightlies/config/install/rosa/ovn-small.json:16]: I didn't get this

[the remark this answers is not in the document]
Comment (krishvoor) [reply dags/openshift_nightlies/config/install/rosa/ovn-small.json:16]: Please update the following:
```
    "openshift_workload_node_instance_type": m5.2xlarge,
    "enable_spot_workers": false
```

File: dags/openshift_nightlies/config/install/rosa/ovn-small.json [hunk 4]
```diff
@@ -12,5 +12,6 @@
     "openshift_network_type": "OVNKubernetes",
     "openshift_worker_instance_type": "m5.2xlarge",
     "machineset_metadata_label_prefix": "machine.openshift.io",
-    "openshift_workload_node_instance_type": null
+    "openshift_workload_node_instance_type": "m5.2xlarge",
+    "enable_spot_workers": false
```
Comment (krishvoor) [inline dags/openshift_nightlies/config/install/rosa/ovn-small.json:16]: Please create a new file: `dags/openshift_nightlies/config/install/rosa/ovn-small-spot.json`
And move the key/values to it.

Comment (athiruma) [reply dags/openshift_nightlies/config/install/rosa/ovn-small.json:16]: if it is the case, I need to add this file to the manifest.yml to read properties of this file.

Comment (krishvoor) [reply dags/openshift_nightlies/config/install/rosa/ovn-small.json:16]: > if it is the case, I need to add this file to the manifest.yml to read properties of this file.

Please update accordingly

Comment (athiruma) [reply dags/openshift_nightlies/config/install/rosa/ovn-small.json]: Done

File: dags/openshift_nightlies/scripts/install/rosa.sh [hunk 5]
```diff
@@ -557,7 +567,15 @@ install(){
         else
             INSTALLATION_PARAMS="${INSTALLATION_PARAMS} --multi-az"  # Multi AZ is default on hosted-cp cluster
         fi
-        rosa create cluster --tags=User:${GITHUB_USERNAME} --cluster-name ${CLUSTER_NAME} --version "${ROSA_VERSION}" --channel-group=${MANAGED_CHANNEL_GROUP} --compute-machine-type ${COMPUTE_WORKERS_TYPE} --replicas ${COMPUTE_WORKERS_NUMBER} --network-type ${NETWORK_TYPE} ${INSTALLATION_PARAMS} ${ROSA_HCP_PARAMS}
+        if [ "$ENABLE_SPOT_WORKERS" == "true" ]; then
+            rosa create cluster --tags=User:${GITHUB_USERNAME} --cluster-name ${CLUSTER_NAME} --version "${ROSA_VERSION}" --channel-group=${MANAGED_CHANNEL_GROUP} --compute-machine-type ${COMPUTE_WORKERS_TYPE} --network-type ${NETWORK_TYPE} ${INSTALLATION_PARAMS} ${ROSA_HCP_PARAMS}
+            _wait_for_cluster_ready ${CLUSTER_NAME}
```
[the remark this answers is not in the document]
Comment (athiruma) [reply dags/openshift_nightlies/scripts/install/rosa.sh:572]: oh, yes right. Will add the checks there.

File: dags/openshift_nightlies/scripts/install/rosa.sh [hunk 6]
```diff
@@ -557,7 +567,15 @@ install(){
...
+            rosa create machinepool -c ${CLUSTER_NAME} --name="${CLUSTER_NAME}-spot-pool" --replicas=${COMPUTE_WORKERS_NUMBER} --instance-type="${COMPUTE_WORKERS_TYPE}" --labels="rosa-spots=true" --use-spot-instances
```
[the remark this answers is not in the document]
Comment (athiruma) [reply dags/openshift_nightlies/scripts/install/rosa.sh:573]: We thought of having total spot instances instead of on-demand.

**Review by krishvoor — CHANGES_REQUESTED** [review]

**Pull request closed without merging**